source/MRCuda/MRCudaBasic.hpp

```diff
@@ -9,6 +12,17 @@ namespace MR
 namespace Cuda
 {
 
+/// spdlog::error the information about some CUDA error including optional filename and line number
+MRCUDA_API void logError( cudaError_t code, const char * file = nullptr, int line = 0 );
+
+/// executes given CUDA function and checks the error code after
+#define CUDA_EXEC( func )\
```

Comment:
Mb define this in debug only? Also it may require cudaDeviceSynchronize

Reply:
error is returned in Release as well, let us check it